hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -1140,7 +1193,17 @@ private void addFileInfoToCompactionLogTable(
       builder.setCompactionReason(compactionReason);
     }
 
-    addToCompactionLogTable(builder.build());
+    CompactionLogEntry compactionLogEntry = builder.build();
+    byte[] key = addToCompactionLogTable(compactionLogEntry);
+    // Add the compaction log entry to the prune queue so that the backup input sst files can be pruned.
+    if (isNativeLibsLoaded) {
+      try {
+        pruneQueue.put(Pair.of(key, compactionLogEntry));
+      } catch (InterruptedException e) {
+        LOG.error("Could not add backup SST files in queue to be pruned. {}",
+            compactionLogEntry.getInputFileInfoList(), e);
+      }
+    }
```

Comment:
I'll suggest moving this code [here](https://github.com/apache/ozone/blob/73e0d2a38399ddb004426e2276c6ff024bc49220/hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java#L815). Otherwise, we will miss new entries from the table.